src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,71 @@
+package duke.task;
+
+import java.util.Objects;
+
+import duke.util.DukeException;
+
+/**
+ * Task Class to store information of task.
+ * @author Jicson Toh
+ */
+public class Task {
+    protected String action;
+    protected boolean isDone;
+    protected String date;
+
+    /**
+     * Creates a new task object.
+     * @param action user input action to be done.
+     * @param isDone check is completed.
+     * @param date date to be completed.
+     * @throws DukeException if action is not specified.
+     */
+    public Task(String action, boolean isDone, String date) throws DukeException {
+        if (Objects.equals(action, "")) {
+            throw new DukeException();
+        }
+        this.action = action;
+        this.isDone = isDone;
+        this.date = date;
+    }
+
+    public String getAction() {
+        return action;
+    }
+
+    public boolean getDone() {
+        return isDone;
+    }
+
+    /**
+     * Checks if the task contains the keyword.
+     * @param keyword input keyword.
+     * @return true if contains the keyword.
+     */
+    public boolean matchKeyword(String keyword) {
```

Comment:
Perhaps try to use prefix such as is, has, was etc for boolean method names.

Reply:
Will take note thanks!